File: tests/fixtures/negative/security/StartActivityWithUntrustedIntent.java
package test;

import android.app.Activity;
import android.content.Intent;

class Screen extends Activity {
    void guarded(String uri, Intent param) throws Exception {
        Intent intent = Intent.parseUri(uri, 0);
        intent.setComponent(null);
        startActivity(intent);
        Intent intent2 = new Intent();
        startActivity(intent2);
        startActivity(param);
    }
}

File: tests/fixtures/negative/security/StartActivityWithUntrustedIntent.kt
package test

import android.app.Activity
import android.content.ComponentName
import android.content.Intent

class Screen : Activity() {
    fun guarded(uri: String, param: Intent) {
        val intent = Intent.parseUri(uri, 0)
        intent.setPackage(packageName)
        startActivity(intent)
        val intent2 = Intent()
        startActivity(intent2)
        startActivity(param)
    }

    fun componentGuard(uri: String) {
        val intent = Intent.parseUri(uri, 0)
        intent.component = ComponentName(packageName, "Target")
        startActivityForResult(intent, 7)
    }
}
